### PR TITLE
Issue/50 streamline jenkins file

### PR DESCRIPTION
**Pull request by kazaamjt**
Created a docker config to run the tests locally and ended up throwing the vagrant stuff out in favor of a docker setup, meanings it's easier (ish) to test locally.
This is purely because of the systemd testing that needed to happen.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -27,8 +27,8 @@ multi_line_output=3
 include_trailing_comma=True
 force_grid_wrap=0
 use_parentheses=True
-line_length=128
+line_length=88
```
Comment (arnaudsjs) [inline setup.cfg]: Why 88?

File: tests/test_file.py [hunk 2]
```diff
@@ -109,5 +109,6 @@ def test_list_files(project):
         """
     )
 
-    out = project.get_stdout()
-    assert out.splitlines() == ["testfile1", "testfile2"]
+    out = project.get_stdout().splitlines()
```
Comment (arnaudsjs) [inline tests/test_file.py]: The assertion in the new implementation is less strict. The test still passes when the output contains more than two elements. Why is this change required?

File: tests/test_resources.py [hunk 3]
```diff
@@ -144,7 +144,7 @@ def test_directory_purge(project, tmpdir, current_state_purged):
 import unittest
 
 host = std::Host(name="server", os=std::linux)
-std::Directory(host=host, path="{test_path_1}", owner="{user}", group="{user}", mode=775, purged=true)
+std::Directory(host=host, path="{test_path_1}", owner="{user}", group="{user}", mode=755, purged=true)
```
Comment (arnaudsjs) [inline tests/test_resources.py]: Why is this required?

File: tests/test_resources.py [hunk 4]
```diff
@@ -294,7 +294,7 @@ def setup(self):
 
 [Install]
 WantedBy=default.target
-""",
+""".encode(),
```
Comment (arnaudsjs) [inline tests/test_resources.py]: Why is this required?

File: tests/test_file.py (hunk 2 above)
Comment (kazaamjt) [reply tests/test_file.py]: Because the order of the elements is not guaranteed.
On my machine out.splitlines() resultes in ["testfile1", "testfile2"], but in the docker container it was quite consistently ["testfile2", "testfile1"].
In addition to this, add_mockfile doesn't clean up it's files between tests and a file from the previous test leaks in. ("testfile")

File: setup.cfg (hunk 1 above)
Comment (kazaamjt) [reply setup.cfg]: while black uses a default line length of 88, but did not seem to apply our specified line length of 128 to import statements. This makes it so that isort is in line with black when it comes to import statements.

File: tests/test_resources.py (hunk 3 above)
Comment (kazaamjt) [reply tests/test_resources.py]: Because at the start of the test, this directory is created with 755.
Here purged is set to true, but an assertion of changes == 1 is made.
Because the mode had been changed from 755 to 775, changes was actually == 2.

File: tests/test_resources.py (hunk 4 above)
Comment (kazaamjt) [reply tests/test_resources.py]: Because it literally will not work otherwise :P
The io function this is passed to expects bytes.

File: Jenkinsfile [hunk 5]
```diff
@@ -2,45 +2,55 @@ pipeline {
...
         script{
-            withCredentials([usernamePassword(credentialsId: 'jenkins_on_openstack', passwordVariable: 'OS_PASSWORD', usernameVariable: 'OS_USERNAME')]) {
-              sh "vagrant up"
-          }
+          sh '''
+          python3 -m venv ${WORKSPACE}/env
+          ${WORKSPACE}/env/bin/pip install --upgrade pip
+          ${WORKSPACE}/env/bin/pip install -r requirements.txt
+          ${WORKSPACE}/env/bin/pip install -r requirements.dev.txt
+          ${WORKSPACE}/env/bin/pip install pytest-inmanta -i ${PIP_INDEX_URL}
```
[the remark this answers is not in the document]
Comment (kazaamjt) [reply Jenkinsfile]: actually, this line can even just be removed (as pytest is run in docker)

[the remark this answers is not in the document]
Comment (arnaudsjs) [reply Jenkinsfile]: The whole build step can even be removed.

[the remark this answers is not in the document]
Comment (kazaamjt) [reply Jenkinsfile]: We still need to install the packages for the lintingstep. Or move linting to the docker file.

File: tests/test_file.py (hunk 2 above)
Comment (arnaudsjs) [reply tests/test_file.py]: * Regarding the order of the elements: You can use the sorted() function to resolve this issue.
* Regarding the resource leak: Let's fix it.

File: Jenkinsfile (hunk 5 above)
[the remark this answers is not in the document]
Comment (arnaudsjs) [reply Jenkinsfile]: Indeed. Overlooked that step.

File: tests/test_resources.py (hunk 3 above)
Comment (arnaudsjs) [reply tests/test_resources.py]: Ok.

File: tests/test_resources.py (hunk 4 above)
Comment (arnaudsjs) [reply tests/test_resources.py]: Indeed. This was a bug in the test case.

File: setup.cfg (hunk 1 above)
Comment (arnaudsjs) [reply setup.cfg]: I was mainly wondering why we choose 88 here, while 128 is used in the inmanta-core.

Comment (kazaamjt) [reply setup.cfg]: It might be an issue in inmanta-core too, but Isort output is currently supressed in core.
see: [issue-1675](https://github.com/inmanta/inmanta/issues/1675)

File: tests/test_file.py (hunk 2 above)
Comment (kazaamjt) [reply tests/test_file.py]: resolved. could you check that the docker steps work for you?

File: Jenkinsfile (hunk 5 above)
[the remark this answers is not in the document]
Comment (kazaamjt) [reply Jenkinsfile]: resolved

File: Jenkinsfile [hunk 6]
```diff
@@ -2,45 +2,55 @@ pipeline {
   agent any
   triggers {
     pollSCM '* * * * *'
-    cron("H H(2-5) * * *")
```
[the remark this answers is not in the document]
Comment (kazaamjt) [reply Jenkinsfile]: Resolved
